indent treats nested function parameters incorrectly

**Tell us about your environment**

* **ESLint Version:** v4.1.1
* **Node Version:** v5.1.0
* **npm Version:** 3.3.12

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1, "parameters": 1} }]*/
```

</details>

**What did you do? Please include the actual source code causing the issue.**
I ran ESlint to check for validation issues.
<!-- Paste the source code below: -->
```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1, "parameters": 1} }]*/

define([
  "bar",
  "baz",
  "qux"
],

function (
  bar,
  baz,
  qux
) {
  qux();
});
```

**What did you expect to happen?**
I expected no validation issues.

**What actually happened? Please include the actual, raw output from ESLint.**
It gave me validation errors:

```
  10:1  error  Expected indentation of 4 spaces but found 2  indent
  11:1  error  Expected indentation of 4 spaces but found 2  indent
  12:1  error  Expected indentation of 4 spaces but found 2  indent
  13:1  error  Expected indentation of 2 spaces but found 0  indent

✖ 4 problems (4 errors, 0 warnings)
  4 errors, 0 warnings potentially fixable with the `--fix` option.
```

If I run `--fix` it ends up looking like this, which to me is double the function parameter spacing that I want (it has 4 spaces instead of 2) and it adds 2 spaces before `) {`.

```js
define([
  "bar",
  "baz",
  "qux"
],

function (
    bar, // These three parameters should have 2 spaces instead of 4
    baz,
    qux
  ) { // This line should have 0 spaces instead of 2
  qux();
});
```

**Note**
Running without the `define` as a simple method works as expected from the [documentation examples](http://eslint.org/docs/rules/indent#functiondeclaration). I.e. this works as expected (no validation issues):

```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1, "parameters": 1} }]*/

function foo(
  bar,
  baz,
  qux
) {
  qux();
}
```

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -750,7 +750,7 @@ module.exports = {
                     const firstTokenOfPreviousElement = previousElement && getFirstToken(previousElement);
 
                     if (previousElement && sourceCode.getLastToken(previousElement).loc.start.line > startToken.loc.end.line) {
-                        offsets.matchIndentOf(firstTokenOfPreviousElement, getFirstToken(element));
+                        offsets.setDesiredOffsets(getTokensAndComments(element), firstTokenOfPreviousElement, 0);
                     }
                 }
             });

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -1472,6 +1472,27 @@ ruleTester.run("indent", rule, {
         },
         {
             code: unIndent`
+                [[
+                ], function(
+                    foo
+                ) {}
+                ]
+            `
+        },
+        {
+            code: unIndent`
+                define([
+                    'foo'
+                ], function(
+                    bar
+                ) {
+                    baz;
+                }
+                )
+            `
+        },
+        {
+            code: unIndent`
                 const func = function (opts) {
                     return Promise.resolve()
                     .then(() => {
@@ -5587,6 +5608,46 @@ ruleTester.run("indent", rule, {
                 [4, 4, 9, "String"],
                 [5, 0, 2, "Punctuator"]
             ])
+        },
+        {
+            code: unIndent`
+                [[
+                ], function(
+                        foo
+                    ) {}
+                ]
+            `,
+            output: unIndent`
+                [[
+                ], function(
+                    foo
+                ) {}
+                ]
+            `,
+            errors: expectedErrors([[3, 4, 8, "Identifier"], [4, 0, 4, "Punctuator"]])
+        },
+        {
+            code: unIndent`
+                define([
+                    'foo'
+                ], function(
+                        bar
+                    ) {
+                    baz;
+                }
+                )
+            `,
+            output: unIndent`
+                define([
+                    'foo'
+                ], function(
+                    bar
+                ) {
+                    baz;
+                }
+                )
+            `,
+            errors: expectedErrors([[4, 4, 8, "Identifier"], [5, 0, 4, "Punctuator"]])
         },
         {
             code: unIndent`

